modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -14,26 +14,86 @@
 
 import ClayAlert from '@clayui/alert';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
-import {useTimeout} from 'frontend-js-react-web';
+import {useIsMounted, useTimeout} from 'frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
-import React, {useEffect, useImperativeHandle, useRef, useState} from 'react';
+import React, {useEffect, useImperativeHandle, useReducer, useRef} from 'react';
 
 import Sidebar from './components/Sidebar';
 
+const initialState = {
+	data: null,
+	error: null,
+	html: null,
+	loading: false,
+	open: true,
+};
+
 const SidebarPanel = React.forwardRef(
 	({fetchURL, onClose, viewComponent: View}, ref) => {
 		const CurrentView = useRef(View);
 
-		const [error, setError] = useState();
-		const [isLoading, setIsLoading] = useState();
-		const [isOpen, setIsOpen] = useState(true);
-		const [resourceData, setResourceData] = useState();
-
 		const delay = useTimeout();
 
+		const isMounted = useIsMounted();
+
+		const dataReducer = (state, action) => {
+			if (isMounted()) {
+				switch (action.type) {
+					case 'CLOSE_SIDEBAR':
+						return {
+							...state,
+							isOpen: false,
+						};
+
+					case 'LOAD_DATA':
+						return {
+							loading: true,
```

Comment:
Missing, `...state` spread here and in a few other places. Your initial state has 5 properties and it should always have 5 properties.